src/Core/Cart/CartRow.php

```diff
@@ -277,13 +277,19 @@ public function processCalculation(Cart $cart)
         $quantity = (int) $rowData['cart_quantity'];
         $this->initialUnitPrice = $this->getProductPrice($cart, $rowData);
 
+        $tools = new Tools();
+
         // store not rounded values, except in round_mode_item, we still need to round individual items
         if ($this->roundType == self::ROUND_MODE_ITEM) {
-            $tools = new Tools();
             $this->initialTotalPrice = new AmountImmutable(
                 $tools->round($this->initialUnitPrice->getTaxIncluded(), $this->precision) * $quantity,
                 $tools->round($this->initialUnitPrice->getTaxExcluded(), $this->precision) * $quantity
             );
+        } elseif ($this->roundType == self::ROUND_MODE_LINE) {
```

Comment:
Hi @eyedmax 

sorry I'm gonna be picky but this mart is too sensitive to accept a modification casually ^^
Are your sure about this modif because it seems like not rounding this as done on purpose considering the comment on L282 `store not rounded values, except in round_mode_item, we still need to round individual items`

Rounding this value too soon could lose some precision in the following computations, and it seems like it's done later L421 in the `applyRound` method. It seems like it is expected the rounding is done in this method instead of `processCalculation` If we start messing with the responsibility of internal methods the code won't make any sense anymore

Reply:
We are currently working on refactoring completely the cart/product computing system because the current system is not stable enough, that's why we are very cautious about modification on this part of the code because we've been proven in the past that even the slightest modification can have massive unexpected side effects 😅

Reply:
@jolelievre yeah, that infamous buttefflylephant effect!